userspace/Makefile

```diff
@@ -23,20 +25,27 @@ KERNEL_OBJ = kernel.o
 .PHONY: all
 all: $(INITRD)
 
-$(INITRD): $(USERSPACE_BINS)
+$(INITRD): $(DUMBFS)
 	$(Q)printf "  SHELL\t scripts/initrd.sh\n"
 	../scripts/initrd.sh
 
+$(DUMBFS):  $(wildcard $(DUMBFS_DATA)/*) $(USERSPACE_BINS)
+	$(Q)printf "  SHELL\t scripts/gen-dumbfs.sh\n"
+	../scripts/gen-dumbfs.sh $(DUMBFS_DATA) $(DUMBFS)
```

Comment:
I think you should merge `gen-dumbfs.sh` in `initrd.sh`.
Btw, `userspace/dumbfs` is unnecessary, we already have `intrd.img` and they are the same.